thecourseforum/theCourseForum2
# Fix course_data.js and instr_data.js missed by eslint in precommit

## Status
- Done

## GitHub Issues addressed (`#IssueNumber`)
- This PR closes #369 

## What I did
- Properly quoted the recursive glob in `precommit`, as suggested in [this SO issue](https://stackoverflow.com/questions/54165756/eslint-glob-is-not-considering-all-directories-recursively)

## Screenshots
- Before
- `precommit` emits no lining errors, even though #369 shows that `course_data.js` and `instr_data.js` have many linting errors
![image](https://user-images.githubusercontent.com/2716970/106929559-7ebeaf00-66e2-11eb-93ed-b9c19bab78d2.png)

- After
- `precommit` emitting linting errors properly for `course_data.js` and `instr_data.js`
![image](https://user-images.githubusercontent.com/2716970/106929479-68185800-66e2-11eb-8a5a-be0850f63852.png)

## Tests
- ~~Run `precommit`, ensure it actually complains~~

The above instructions worked before I refactored `course_data.js` and `instr_data.js` to no longer offend eslint. Now, running `./precommit` (or testing GitHub Actions) has no errors.

Testing that `course_data.js` and `instr_data.js` are actually linted:
1. Edit `course_data.js` or `instr_data.js` to trigger an `eslint` message, such as by adding `var x = 5;` in the body somewhere and not using the variable.
2. Run `./precommit`, ensure that a message is raised by eslint.
3. Test GitHub Actions locally, if you have the capacity. Make sure `eslint` fails.
4. Undo your lint-breaking changes to `course_data.js` and `instr_data.js`


## Questions/Discussions/Notes
- ~~Someone will have to fix/silence the linter errors happening in those files, as there are quite a lot of them...~~
  - Done! 

## To-do
[x] Fix the quoting in `./precommit`
[x] Fix the quoting in GitHub actions
[x] Fix / silence the linter errors in the offending JS files

## Merging the PR
- Who should merge this PR?
- Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
